### ISSUE TITLE
unexpected-keyword-arg in dataclasses with kw_only=True, inheritance and defaults

### ISSUE THREAD

**Issue by sharhalakis** [PyCQA/pylint#6550]
### Bug description

This is practically #6275 when using the new kw_only=True

The following code causes the bogus error. The key points are:

- the kw_only argument on class A 
- the presence of a default in class A 
- that class B includes a value without a default.

The problem doesn't happen if there's no default in-between or if inheritance isn't used.

```python
# pylint: disable=missing-module-docstring,invalid-name,too-few-public-methods,missing-class-docstring

import dataclasses as dc

@dc.dataclass(kw_only=True)
class A:
    a: str
    b: int = 1

@dc.dataclass(kw_only=True)
class B(A):
    c: str

c = B(a='aaa', c='ccc')
```

### Command used

```shell
pylint b.py
```

### Pylint output

```shell
$ pylint b.py 
************* Module b
b.py:15:4: E1123: Unexpected keyword argument 'c' in constructor call (unexpected-keyword-arg)

------------------------------------------------------------------
Your code has been rated at 3.75/10 (previous run: 3.75/10, +0.00)
```

### Expected behavior

No pylint errors. This is valid code.

### Pylint version

```shell
$ pylint --version
pylint 2.12.2
astroid 2.9.3
Python 3.10.4 (main, Mar 24 2022, 13:07:27) [GCC 11.2.0]
```

### OS / Environment

```
$ lsb_release -a
No LSB modules are available.
Distributor ID: Debian
Description:    Debian GNU/Linux bookworm/sid
Release:        testing
Codename:       bookworm
```

**Comment by anis-campos** [comment]
+1. `kw_only` is very useful and the completely disabling `unexpected-keyword-arg` seems counter productive.

So for now we a re disabling the linter on every instantiation.

**Comment by smrnff** [comment]
+1 We really need the feature too. 

> So for now we are disabling the linter on every instantiation.

We do the same things. It freaks me out.

**Comment by DanielNoord** [comment]
Fixed with in https://github.com/PyCQA/astroid/pull/1764

@jacobtylerwalls Do we want test for this and for the original issue that that PR targeted in `pylint`? Or is `astroid` enough in your opinion?

**Comment by jacobtylerwalls** [comment]
Let's merge at least one test case in pylint, since the astroid one is a little convoluted (on purpose, to catch the string contains issue.)

**Comment by DanielNoord** [comment]
Sure! I'll open a PR for you to review 😄

**Comment by yarinbar** [comment]
I am experiencing the same issue even when i dont use default values and I even tried defining my own init function:

```python
@dataclass(kw_only=True)
class Alert:
    """of alerts"""
    alert_type: str
    location: torch.Tensor
    # score: float = 1.

    def __init__(self, **kwargs):
        for k, v in kwargs.items():
            if k != 'self':
                setattr(self, k, v)


@dataclass(kw_only=True)
class RoofingAlert(Alert):
    # type = 'roofing'
    # location: torch.Tensor
    angle: float
    cath_direction: torch.Tensor
    vess_direction: torch.Tensor

    def __init__(self, **kwargs):
        super().__init__(alert_type=kwargs.get('alert_type'), location=kwargs.get('location'))
        for k, v in kwargs.items():
            if k != 'self':
                setattr(self, k, v)
```

**Comment by Pierre-Sassoulas** [comment]
Hello and thank you for reporting an issue @yarinbar. Could you open a new issue, this one is closed and the comment is going to get lost.
